### PR TITLE
Added dependencies for System.CommandLine, NuGet.Client, Microsoft.Extensions.Logging

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
https://github.com/dotnet/templating/issues/3379 
fixes https://github.com/dotnet/templating/issues/3104

### Solution
Adds dependencies for:
- System.CommandLine
- NuGet.Client
- Microsoft.Extensions.Logging

Unified all versions in Versions.props file.
Test packages versions are given in https://github.com/dotnet/templating/blob/main/test/Directory.Build.targets

### CODE REVIEW

**Comment by vlada-shubina** [comment]
Subscriptions to be added once merged to main

**Comment by DavidKarlas** [comment]
Is this fixing https://github.com/dotnet/templating/issues/3104 ?

**Review by bekir-ozturk — APPROVED** [review]

**Comment by vlada-shubina** [comment]
> Is this fixing #3104 ?

not really - this is only for dependencies available through arcade. There might be something else, that mentions versions directly. I'll check

**Comment by vlada-shubina** [comment]
> > Is this fixing #3104 ?
> 
> not really - this is only for dependencies available through arcade. There might be something else, that mentions versions directly. I'll check

I extracted remaining versions to Versions.props. 

Test only dependencies remain in https://github.com/dotnet/templating/blob/main/test/Directory.Build.targets

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -15,6 +15,34 @@
       <Uri>https://github.com/dotnet/runtime</Uri>
       <Sha>cf2938f44fc1494ae984c52a1568a03d51e513ca</Sha>
     </Dependency>
```
Comment (MichaelSimons) [inline eng/Version.Details.xml:17]: Can a dependency by added for Microsoft.DotNet.Cli.CommandLine as well?  This is going to be needed for source-build.

**Review by MichaelSimons — APPROVED** [review]
One comment, rest looks good.

**Comment by mmitche** [comment]
@vlada-shubina Are the new Microsoft.Extensions.* dependencies for product functionality, or for testing?

**Comment by vlada-shubina** [comment]
@mmitche we use `Microsoft.Extensions.Logging` as logging solution in template engine core, so it's product.

**Pull request merged**